mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala

```diff
@@ -269,6 +270,8 @@ class LDA private (
    */
   @Since("1.3.0")
   def setCheckpointInterval(checkpointInterval: Int): this.type = {
+    require(checkpointInterval > 0,
+      s"Period between checkpoints must be positive but got ${checkpointInterval}")
```

Comment:
The legal bounds of checkpointInterval is >= 1 or = -1, -1 means disable checkpoint.